common/maplib/move-rule.mjs

```diff
@@ -0,0 +1,51 @@
+import {createRequire} from 'module';
+const require = createRequire(import.meta.url);
+
+const config = require('config');
+const movingRequestThreshold = config.get('movingRequestThreshold');
+
+class MoveRule{
+
+  _getManhattanDistance(a,b){
+    return Math.abs(a.x - b.x) + Math.abs(a.y - b.y);
+  }
```

Comment:
Well ... I think you may want to return `inf` if two coordinates are in different maps.
In this way, we do not need same map check since it will be covered in speed check.

Reply:
`MapCoord.distanceTo(anotherMapCoord)`
`distanceBetween(aCoord, bCoord)`
These are two function that you may want to implement in `map.mjs`.